pkg/certificate/manager.go

```diff
@@ -152,9 +154,83 @@ func (m *Manager) handleMRCEvent(mrcClient MRCClient, event MRCEvent) error {
 			m.mu.Unlock()
 		}
 	case MRCEventUpdated:
-		// TODO
+		oldMRC := event.OldMRC
+		newMRC := event.NewMRC
+
+		// check if the MRC's intent has been updated from active to passive and the
+		// certificate statuses are all issuing
+		if oldMRC.Intent != newMRC.Intent && newMRC.Intent == constants.MRCIntentPassive &&
```

Comment:
is the idea that there will be multiple if statements here, for each stage change, each one creating a reconciler?

Can you add a comment that might show what all of those checks might be?

Reply:
The checks for MRCUpdated events are roughly outlined in this doc https://docs.google.com/document/d/1QF7wL1su4CUzwo3Yo2_vFL90XG2d-wUhBU6uifNrkws/edit#, but are more specifically outlined in the following issues:
- #5091 
- #5089
- #5092

I can also add a comment.

Reply:
and i'm guessing it also performs the actual update via the compute client? So it seems the job of the update is split between the cert manager, which has several checks for switching on status case, and the `handleActiveToPassiveMRCStatusChange`, which *actually* also performs the update. So that's a bit confusing to me.

It seems the mrcReconciler would likely want to reconcile both the old and the new mrc, and not just the new one? Or is the idea that there can be multiple reconciler objects running simultaneously?

If that's the case there needs to be extra protections to make sure that one mrc doesn't race ahead while another gets stuck behind.

Finally i go back to my original argument: why is this logic spread across 2 objects? Why not put the entire event loop/handleActiveToPassiveMRCStatusChange on a new reconciler object, and have the reconciler object be long lived? Why does it need to be a temporal object that finishes upon completion? Instead of temporal object, it can be a goroutine on a more permanent object?

Reply:
with that, instead of having an updateFunc variable, it can be 

```
func (r *reconciler) reconcile(ctx context.Context, oldName, newName string) {
   for range ticker.C {
    oldMRC, newMRC := get(oldName), get(newName)
    switch{
     case isActive(oldMRC, newMRC):
          r.updateActive(oldMRC, newMRC)
     case isPassive(oldMRC, newMRC):
          r.updatePassive(oldMRC, newMRC)
      .....
    }
}
```


That's a really rough outline, but i think sets the stage.